mlos_bench/mlos_bench/tests/storage/exp_load_test.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Unit tests for the storage subsystem.
+"""
+
+from mlos_bench.storage import Storage
+from mlos_bench.tunables import TunableGroups
+from mlos_bench.environment import Status
+
+# pylint: disable=redefined-outer-name
+
+
+def test_exp_load_empty(exp_storage_memory_sql: Storage.Experiment):
+    """
+    Try to retrieve old experimental data from the empty storage.
+    """
+    (configs, scores) = exp_storage_memory_sql.load()
+    assert not configs
+    assert not scores
+
+
+def test_exp_pending_empty(exp_storage_memory_sql: Storage.Experiment):
+    """
+    Try to retrieve pending experiments from the empty storage.
+    """
+    trials = list(exp_storage_memory_sql.pending())
+    assert not trials
+
+
+def test_exp_trial_pending(exp_storage_memory_sql: Storage.Experiment,
+                           tunable_groups: TunableGroups):
+    """
+    Start a trial and check that it is pending.
+    """
+    trial = exp_storage_memory_sql.trial(tunable_groups)
+    (pending,) = list(exp_storage_memory_sql.pending())
+    assert pending.trial_id == trial.trial_id
+    assert pending.tunables == tunable_groups
+
+
+def test_exp_trial_pending_many(exp_storage_memory_sql: Storage.Experiment,
+                                tunable_groups: TunableGroups):
+    """
+    Start THREE trials and check that both are pending.
+    """
+    config1 = tunable_groups.copy().assign({'rootfs': 'ext4'})
+    config2 = tunable_groups.copy().assign({'rootfs': 'ext2'})
+    trial_ids = {
+        exp_storage_memory_sql.trial(config1).trial_id,
+        exp_storage_memory_sql.trial(config2).trial_id,
+        exp_storage_memory_sql.trial(config2).trial_id,  # Submit same config twice
+    }
+    pending_ids = {pending.trial_id for pending in exp_storage_memory_sql.pending()}
+    assert len(pending_ids) == 3
+    assert trial_ids == pending_ids
```

Comment:
Could add a check here that there's still only two configs as well in the DB.

Reply:
same as the above:
> Storage API currently does not have any methods to retrieve the individual configs, and I am not sure I want to expose that functionality on the mlos_bench side... For unit tests, however, I can tap into SQLite directly and check if the data is there. Let's discuss